torch/testing/_internal/common_nn.py

```diff
@@ -5394,6 +5394,24 @@ def single_batch_reference_criterion_fn(*args):
     # reduction is 'sum' or 'mean' which results in a scalar
     return output
 
+def single_batch_reference_cosineembeddingloss(*args):
```

Comment:
I think it's possible to make `single_batch_reference_criterion_fn` slightly more generic so it can handle `CosineEmbeddingLoss` as well. I mention this because there are a few other non-standard criterions that will need tests soon and it'd be nice to use the same reference fn for those too